OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -222,6 +221,24 @@ const OpenSim::PhysicalFrame& Joint::getParentFrame() const
     return getConnector<PhysicalFrame>("parent_frame").getConnectee();
 }
 
+Coordinate::MotionType Joint::getMotionType(CoordinateIndex cix) const
+{
+    OPENSIM_THROW_IF(cix >= _motionTypes.size(), Exception,
+        "Joint::getMotionType() given an invalid CoordinateIndex");
+    return _motionTypes[cix];
+}
+
+void Joint::setMotionType(CoordinateIndex cix, Coordinate::MotionType mt)
+{
+    OPENSIM_THROW_IF(cix >= numCoordinates(), Exception,
+        "Joint::setMotionType() for an invalid CoordinateIndex");
+    if (_motionTypes.size() <= cix)
```

Comment:
Then perhaps the following line should be `_motionTypes.resize(cix+1);`? What doesn't make sense to me is why the `if` statement catches situations where `_motionTypes.size() <= cix` but the result is to increase the size to `numCoordinates()`. The resulting action doesn't immediately follow from the logic of the test. Perhaps a comment could be added to explain what's going on.